src/utils/storage.js

```diff
@@ -0,0 +1,38 @@
+// @ts-check
+
+import { openDB } from 'idb'
+
+const DEFAULT_IDB_VERSION = 1
+const DEFAULT_IDB_STORAGE_NAME = 'saturn-db'
+const DEFAULT_SATURN_STORAGE_NAME = 'saturn-client'
+
+/**
+ * @typedef {object} Storage
+ * @property {function():boolean} check - Checks if the provided Storage is accessible
+ * @property {function(string):Promise<any>} get - Retrieves the value associated with the key.
+ * @property {function(string,any):Promise<void>} set - Sets a new value for the key.
+ * @property {function(string):Promise<any>} delete - Deletes the value associated with the key.
+ */
+
+/**
+ * @function indexedDbStorage
+ * @returns {Storage}
+ */
+export function indexedDbStorage () {
+  const indexedDbExists = (typeof window !== 'undefined') && window?.indexedDB
+  let dbPromise
+  if (indexedDbExists) {
+    dbPromise = openDB(DEFAULT_IDB_STORAGE_NAME, DEFAULT_IDB_VERSION, {
+      upgrade (db) {
+        db.createObjectStore(DEFAULT_SATURN_STORAGE_NAME)
+      }
+    })
+  }
+
+  return {
+    check: () => Boolean(indexedDbExists),
```

Comment:
I dont think this method is necessary. the user should know which environment the js-client is running in. In other words, this check should be run before this storage class is even created.

If `indexedDbExists` is false then every method will silently fail, which is confusing.

Reply:
the reasoning for adding this is to safeguard against dumb clients. 

If someone wants to implement a new storage interface or we update an existing one and it drops and results in some reduced support, wanted to still make sure that the software works by always checking that the injected storage is working as intended. 

I do agree that the callee should still write checks, but wondering if adding an explicit `check` reduces the likelihood of our software failing.